src/lib/services/addon-service.test.ts

```diff
@@ -73,7 +74,7 @@ test('should not allow addon-config for unknown provider', async () => {
             },
             'test',
         );
-    }).rejects.toThrow(new TypeError('Unknown addon provider unknown'));
+    }).rejects.toThrow(BadDataError);
```

Comment:
Here, I'm wondering if we shouldn't have fixed the src/lib/routes/util.ts handleError match statement instead to return 400 on TypeErrors.

Reply:
Resolved!